File: BEmuJava/bemu/src/main/java/com/bemu/BEmu/HistoricalDataRequest/ElementHistoricInt.java
//------------------------------------------------------------------------------
// <copyright project="BEmuJava" file="BEmu.HistoricalDataRequest.ElementHistoricInt.java" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

package com.bemu.BEmu.HistoricalDataRequest;

import com.bemu.BEmu.Name;

public class ElementHistoricInt extends ElementParent
{
	private final int _value;
	private final String _name;
	
	ElementHistoricInt(String name, int value)
	{
		this._name = name;
		this._value = value;
	}
	
    public Name name()
    {
    	return new Name(this._name);
    }
    
    public int numValues()
    {
    	return 1;
    }
    
    public int numElements()
    {
    	return 0;
    }
    
    public boolean isComplexType()
    {
    	return false;
    }
    
    public boolean isArray()
    {
    	return false;
    }
    
    public boolean isNull()
    {
    	return false;
    }
    
    protected StringBuilder prettyPrint(int tabIndent)
    {
    	return super.prettyPrintHelper(tabIndent, String.valueOf(this._value));
    }
    
    public Object getValue()
    {
    	return this._value;
    }
    
    public Object getValue(int index) throws Exception
    {
    	if(index == 0)
    		return this.getValue();
    	else
    		return super.getValue();
    }
    
    public int getValueAsInt32()
    {
        return this._value;
    }

    public int getValueAsInt32(int index) throws Exception
    {
        if (index == 0)
            return this.getValueAsInt32();
        else
            return super.getValueAsInt32(index);
    }
    
    public long getValueAsInt64()
    {
        return this._value;
    }

    public long getValueAsInt64(int index) throws Exception
    {
        if (index == 0)
            return this.getValueAsInt64();
        else
            return super.getValueAsInt64(index);
    }
    
    public float getValueAsFloat32()
    {
        return this._value;
    }

    public float getValueAsFloat32(int index) throws Exception
    {
        if (index == 0)
            return this.getValueAsFloat32();
        else
            return super.getValueAsFloat32(index);
    }
    
    public double getValueAsFloat64()
    {
        return this._value;
    }

    public double getValueAsFloat64(int index) throws Exception
    {
        if (index == 0)
            return this.getValueAsFloat64();
        else
            return super.getValueAsFloat64(index);
    }
    
    public String getValueAsString()
    {
        return String.valueOf(this._value);
    }
}

File: BEmu_cpp/src/HistoricalDataRequest/ElementHistoricInt.cpp
//------------------------------------------------------------------------------
// <copyright project="BEmu_cpp" file="src/HistoricalDataRequest/ElementReferenceInt.cpp" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

#include "HistoricalDataRequest/ElementHistoricInt.h"

#include "BloombergTypes/Name.h"

namespace BEmu
{
	namespace HistoricalDataRequest
	{
		ElementHistoricInt::ElementHistoricInt(const std::string& name, int value)
		{
			this->_name = name;
			this->_value = value;
		}

		ElementHistoricInt::~ElementHistoricInt()
		{
		}


		Name ElementHistoricInt::name() const
		{
			Name result(this->_name.c_str());
			return result;
		}

		size_t ElementHistoricInt::numValues() const
		{
			return 1;
		}

		size_t ElementHistoricInt::numElements() const
		{
			return 0;
		}

		
		bool ElementHistoricInt::isArray() const
		{
			return false;
		}

		bool ElementHistoricInt::isComplexType() const
		{
			return false;
		}


		int ElementHistoricInt::getValueAsInt32(int index) const
		{
			if(index == 0)
				return this->_value;
			else
				throw elementPtrEx;
		}

		long ElementHistoricInt::getValueAsInt64(int index) const
		{
			if(index == 0)
				return (long)this->_value;
			else
				throw elementPtrEx;
		}

		float ElementHistoricInt::getValueAsFloat32(int index) const
		{
			if(index == 0)
				return (float)this->_value;
			else
				throw elementPtrEx;
		}

		double ElementHistoricInt::getValueAsFloat64(int index) const
		{
			if(index == 0)
				return (double)this->_value;
			else
				throw elementPtrEx;
		}

		const char * ElementHistoricInt::getValueAsString(int index) const
		{
			if(index == 0)
				return ElementPtr::toCharPointer(this->_value);
			else
				throw elementPtrEx;
		}


		std::ostream& ElementHistoricInt::print(std::ostream& stream, int level, int spacesPerLevel) const
		{
			ElementPtr::prettyPrintHelper(stream, level, spacesPerLevel, this->_value);
			return stream;
		}

	}
}